crates/uv-interpreter/src/discovery.rs

```diff
@@ -378,52 +391,61 @@ fn python_interpreters<'a>(
                 .inspect_err(|err| debug!("{err}")),
             Err(err) => Err(err),
         })
-        .filter(move |result| match result {
+        .filter(move |result| {
+            // Always skip shim environments that are not virtual environments; this filter is deferred
+            // from `python_executables` to avoid querying the interpreter there
```

Comment:
What is a shim environment that isn't a virtual environment? What would appear here to fail this filter?

Reply:
Since we're just taking the first executable and calling it a "shim environment" in `python_executables` this filtering is where we check if the executable is _actually_ a shim which invokes inside a virtual environment.

Reply:
(this and the other review comment are my concern with this implementation, it's misleading but we do it to avoid querying the interpreter during `python_executables`)